config/src/config/state_sync_config.rs

```diff
@@ -11,7 +11,11 @@ use serde::{Deserialize, Serialize};
 use serde_yaml::Value;
 
 // The maximum message size per state sync message
-const MAX_MESSAGE_SIZE: usize = 10 * 1024 * 1024; /* 10 MiB */
+const SERVER_MAX_MESSAGE_SIZE: usize = 10 * 1024 * 1024; // 10 MiB
+
+// The maximum message size per state sync message (for v2 data requests)
+const CLIENT_MAX_MESSAGE_SIZE_V2: usize = 15 * 1024 * 1024; // 15 MiB (used for v2 data requests)
+const SERVER_MAX_MESSAGE_SIZE_V2: usize = 40 * 1024 * 1024; // 40 MiB (used for v2 data requests)
```

Comment:
Server can send up to 40 MB, but client can accept upto 15 MB?
Shouldn't it be reverse?

Reply:
The minimum of these values is used. i.e., the server will send at-most 40MB (even if the client requests more). The server will otherwise respect the client specified value 😄